keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourcePersister.kt

```diff
@@ -25,6 +26,18 @@ class ResourcePersister(
   private val clock: Clock,
   private val publisher: ApplicationEventPublisher
 ) {
+  fun upsert(resource: SubmittedResource<Any>): Resource<out Any> =
+    handlers.supporting(resource.apiVersion, resource.kind)
+      .normalize(resource)
+      .also {
+        try {
+          val existing = resourceRepository.get(it.name, Any::class.java)
+          return update(existing.name, resource)
+        } catch (e: NoSuchResourceException) {
+          return create(resource)
```

Comment:
Does it make sense to propagate the upsert concept all the way to the `ResourceRepository`? Alternatively having an `isThisRegistered(name) : Boolean` method. Catching exceptions feels bad.

Reply:
(as discussed offline) I'll do a method for now.